torch/csrc/distributed/rpc/rpc_agent.cpp

```diff
@@ -15,6 +15,20 @@ const WorkerInfo& RpcAgent::getWorkerInfo() const {
   return workerInfo_;
 }
 
+std::shared_ptr<RpcAgent> RpcAgent::defaultRpcAgent_ = nullptr;
+
+std::shared_ptr<RpcAgent> RpcAgent::getDefaultRpcAgent() {
```

Comment:
This method is called very frequently. I am wondering whether it should return a reference to avoid creating a new `shared_ptr` on every call.

Reply:
I just realized we might run into a use-after-free problem if we return a reference here. The reason is that the shared_ptr might actually get freed while another thread holds a reference to it. We can address this problem when we convert all of our static objects to pointers as suggested in https://github.com/pytorch/pytorch/issues/27412.